data_structures/stacks/infix_to_postfix_conversion.py

```diff
@@ -50,9 +77,26 @@ def infix_to_postfix(expression_str: str) -> str:
                 postfix.append(stack.pop())
             stack.pop()
         else:
-            while not stack.is_empty() and precedence(char) <= precedence(stack.peek()):
-                postfix.append(stack.pop())
-            stack.push(char)
+            while True:
+                if stack.is_empty():
+                    stack.push(char)
+                    break
+
+                char_precedence = precedence(char)
+                tos_precedence = precedence(stack.peek())
+
+                if char_precedence > tos_precedence:
+                    stack.push(char)
+                    break
+                elif char_precedence == tos_precedence:
+                    if associativity(char) == "RL":
+                        stack.push(char)
+                        break
+                    else:
+                        postfix.append(stack.pop())
+                else:
+                    postfix.append(stack.pop())
```

Comment:
Could refactor this to avoid some nesting:
```suggestion
                if char_precedence > tos_precedence:
                    stack.push(char)
                    break
                if char_precedence < tos_precedence:
                    postfix.append(stack.pop())
                    break
                # Precedences are equal
                if associativity(char) == "RL":
                    stack.push(char)
                    break
                postfix.append(stack.pop())
```